controllers/event/eventbus_controller.go

```diff
@@ -0,0 +1,114 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	openfunctionevent "github.com/openfunction/apis/event/v1alpha1"
+	"github.com/openfunction/controllers/event/adapters"
+	"github.com/openfunction/pkg/util"
+)
+
+const DefaultEventBusName = "default"
+
+// EventBusReconciler reconciles a EventBus object
+type EventBusReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	ctx    context.Context
+}
+
+//+kubebuilder:rbac:groups=event.openfunction.io,resources=eventbus,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=event.openfunction.io,resources=eventbus/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=event.openfunction.io,resources=eventbus/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the EventBus object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *EventBusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.ctx = ctx
+	log := r.Log.WithValues("EventBus", req.NamespacedName)
+	log.Info("eventbus reconcile starting...")
+
+	var eb openfunctionevent.EventBus
+
+	if err := r.Get(ctx, req.NamespacedName, &eb); err != nil {
+		log.V(1).Info("EventBus deleted", "error", err)
+		return ctrl.Result{}, util.IgnoreNotFound(err)
+	}
+
+	if _, err := r.createOrUpdateEventBus(&eb); err != nil {
+		log.Error(err, "Failed")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// 1. Generate dapr component specs for the event bus
+// 2. Create dapr component by using specs above
+// 3. Set the annotation to assign the name of the dapr component associated with the event bus to "component-name"
+func (r *EventBusReconciler) createOrUpdateEventBus(eb *openfunctionevent.EventBus) (ctrl.Result, error) {
+	log := r.Log.WithName("createOrUpdate")
+
+	if eb.Spec.Topic == "" {
+		eb.Spec.Topic = DefaultEventBusName
+	}
+
+	// Converts the event bus specification into a connector
+	// containing the specification for creating the dapr component.
+	c, err := adapters.NewEventBusConnectors(eb)
+	if err != nil {
+		log.Error(err, "Failed to generate connectors")
+		return ctrl.Result{}, err
+	}
+
+	// Create dapr component to adapter the event bus
+	if err := CreateOrUpdateDaprComponent(r.Client, r.Scheme, r.Log, c, eb); err != nil {
+		log.Error(err, "Failed to create dapr component", "namespace", eb.Namespace, "name", eb.Name)
+		return ctrl.Result{}, err
+	}
+
+	eventBusAnnotations := eb.GetAnnotations()
+	eventBusAnnotations["component-name"] = c.Component.Name
```

Comment:
We'd better to use hash to check whether the crd has been changed to avoid loop

https://github.com/OpenFunction/OpenFunction/blob/main/controllers/function_controller.go#L117